src/edit/index.vue

```diff
@@ -1,9 +1,116 @@
+        {{ infoMessage }}
+      </tce-overlay>
+      <div ref="player" class="player"></div>
+    </div>
+  </div>
 </template>
 
 <script>
+import createUpload from '../createUpload';
+import { ELEMENT_STATE } from '../shared';
+import ElementPlaceholder from '../tce-core/ElementPlaceholder.vue';
+import get from 'lodash/get';
+import TceOverlay from './Overlay.vue';
+
+const DEFAULT_ERROR_MSG = 'Something went wrong.';
+const UPLOADING_MSG = 'Video is uploading... Do not leave the page.';
+const PROCESSING_MSG = 'Video is processing...';
+
 export default {
-  name: 'tce-api-video'
+  name: 'tce-api-video',
+  inject: ['$elementBus'],
+  props: {
+    element: { type: Object, required: true },
+    isFocused: { type: Boolean, default: false },
+    isDisabled: { type: Boolean, default: false }
+  },
+  data: () => ({ file: null }),
+  computed: {
+    videoId: ({ element }) => element.data?.videoId,
+    uploadUrl: ({ element }) => element.data?.uploadUrl,
+    embedCode: ({ element }) => element.data?.embedCode,
+    fileName: ({ element }) => element.data?.fileName,
+    playable: ({ element }) => element.data?.playable,
+    status: ({ element }) => element.data?.status,
+    error: ({ element }) => element.data?.error,
+    didUploadFail: ({ status, file }) => status === ELEMENT_STATE.UPLOADING && !file,
+    isEmpty: ({ error, fileName }) => !error && !fileName,
+    infoMessage: ({ error, status, playable, didUploadFail }) => {
+      if (error || didUploadFail) return;
+      if (status === ELEMENT_STATE.UPLOADING) return UPLOADING_MSG;
+      if (!playable) return PROCESSING_MSG;
+    },
+    isPreparedToUpload: ({ videoId, file, uploadUrl }) => videoId && file && uploadUrl
+  },
+  methods: {
+    appendVideo() {
+      const { player } = this.$refs;
+      if (!player) return;
+      player.innerHTML = this.embedCode;
+    },
+    upload() {
+      const { videoId, file, uploadUrl: url } = this;
+      createUpload({ videoId, file, url })
+        .then(() => {
+          this.file = null;
+          this.$emit('save', { ...this.element.data, status: ELEMENT_STATE.UPLOADED });
+        })
+        .catch(err => this.$elementBus.emit('error', err.response));
+    }
+  },
+  watch: {
+    embedCode: 'appendVideo',
+    videoId() {
+      if (this.isPreparedToUpload) this.upload();
+    }
+  },
+  mounted() {
+    this.appendVideo();
+
+    this.$elementBus.on('save', ({ file }) => {
+      this.file = file;
+      this.$emit('save', {
+        ...this.element.data,
+        fileName: file.name,
+        status: ELEMENT_STATE.UPLOADING
+      });
+    });
+
+    this.$elementBus.on('error', ({ data }) => {
+      this.$emit('save', {
+        ...this.element.data,
+        error: get(data, 'error.message', DEFAULT_ERROR_MSG)
+      });
+    });
+  },
+  components: { ElementPlaceholder, TceOverlay }
 };
 </script>
+
+<style lang="scss" scoped>
+.tce-api-video {
+  position: relative;
+}
+
+.player {
+  height: 25.625rem;
```

Comment:
How does this behave on large screens? Should we make it scale fluidly by stretching it to `max-height`?

Reply:
It behaves fine on large screens. The iframe is responsive and fits the given height.